tests/load/filesystem/test_credentials_mixins.py

```diff
@@ -170,15 +216,22 @@ def test_aws_object_store_rs_credentials(driver: str, fs_creds: Dict[str, Any])
         region_name=fs_creds["region_name"],
     )
     assert creds.aws_session_token is not None
-    object_store_rs_creds = creds.to_object_store_rs_credentials()
-    assert object_store_rs_creds["aws_session_token"] is not None
-    assert can_connect(AWS_BUCKET, object_store_rs_creds)
+    assert can_connect(AWS_BUCKET, creds, mixin)
+    if mixin == WithObjectStoreRsCredentials:
+        object_store_rs_creds = creds.to_object_store_rs_credentials()
+        assert object_store_rs_creds["aws_session_token"] is not None
 
 
 @pytest.mark.parametrize(
     "driver", [driver for driver in ALL_FILESYSTEM_DRIVERS if driver in ("gs")]
 )
-def test_gcp_object_store_rs_credentials(driver, fs_creds: Dict[str, Any]) -> None:
+@pytest.mark.parametrize("mixin", ALL_CREDENTIALS_MIXINS)
+def test_gcp_credentials_mixins(
+    driver, fs_creds: Dict[str, Any], mixin: Type[TCredentialsMixin]
+) -> None:
+    if mixin == WithPyicebergConfig:
+        pytest.skip("`WithPyicebergConfig` mixin currently not implemented for GCP.")
```

Comment:
Seems like `pyiceberg` only accepts OAuth authentication for GCP. I tried it with the OAuth client used for BigQuery tests (the one in `[destination.bigquery.credentials]`), but got a permissions error. Perhaps that client doesn't have read permissions on `gs://ci-test-bucket`. If we have (or create) an Oauth client with correct permissions, I can test it again and merge the code for GCP.

Reply:
Changed scopes on OAuth client. Works now.